src/screens/DiscoveryScreenV2/elements/Search.js

```diff
@@ -36,7 +36,8 @@ const DiscoverySearch = ({
   fetchDiscoveryData = () => {},
   fetchData,
   onCancelToken = () => {},
-  hideBackIcon = false
+  hideBackIcon = false,
+  autoFocus = true
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

A new prop `autoFocus` has been added to the component. Make sure it's used properly throughout the component and its default value is set appropriately.
```diff
-  hideBackIcon = false
+  hideBackIcon = false,
+  autoFocus = true
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->